### DISCUSSION TITLE
The highlight on vscode seems be wrong

### DISCUSSION THREAD

**Discussion by ws2bhhh** [clangd/clangd#1755] [Q&A] [answered]
cland version: 16.0.2
I write some code in C++. This problem won't happend when I changed the file type to .c
Wrong highlight at Line 151, Line 162
<img width="1018" alt="help" src="https://github.com/clangd/clangd/assets/25045666/fc42005f-3f1a-4c71-a87a-18fe07d31f0b">

**Comment by HighCommander4** [comment]
I think that's likely to be an issue with client-side highlighting or another extension.

To be sure, can you try disabling the clangd plugin and seeing what the highlighting looks like then?

**Reply by ws2bhhh** [reply]
After disable clangd plugin:
<img width="1233" alt="help2" src="https://github.com/clangd/clangd/assets/25045666/dd3e8a99-6c8b-40e7-8c0e-355e82c94a55">

**Reply by ws2bhhh** [reply, marked as answer]
Progress: this seems to have been fixed after removing the .clang-tidy file.
Progress: Instead of deleting .clang-tidy, removing "modernize-*,\" from .clang-tidy seems to fix the problem.
.clang-tidy:
[clang-tidy.txt](https://github.com/clangd/clangd/files/12495639/clang-tidy.txt)

**Reply by HighCommander4** [reply]
Interesting. I wonder if your editor is configured to style diagnostic range with a foreground color rather than the more typical squiggly underline.

If you enable the `modernize-*` diagnostics, are there diagnostics issues for the affected lines (such as that `else` token)? You should be able to check by hovering over the token, the hover should show the diagnostic if there is one.

**Reply by ws2bhhh** [reply]
I am not configured to style the diagnostic range with a foreground color.

With `modernized-*` diagnostics enabled, everything behaves normally. More specifically, all hover->diagnostics behaves normally, except for incorrect highlighting.

**Reply by ws2bhhh** [reply]
I'm afraid that's as far as it goes. It appears to be just a random, low impact failure. I tried to give minimal reproducible code, but none of them had a similar problem.
Using a lot of C code by extern "C" as in "test2.cpp" above did not succeed in reproducing a similar problem.
Attempts to reduce the total amount of code in "test2.cpp" were also unsuccessful, as I was unable to eliminate irrelevant parts ("test2.cpp" uses easyX, a C++ only graphic library, and uses mostly C code for the main logic). 
When deleting code from "test2.cpp", the error-highlighted positions change slightly (even if some empty lines are removed).

**Reply by HighCommander4** [reply]
One diagnostic technique that could shed some light is: with the editor caret over an incorrectly highlighted token, run the vscode command "Developer: Inspect Editor Tokens and Scopes". That will show a popup with various information related to coloring, feel free to post a screenshot of that.

**Reply by ws2bhhh** [reply]
<img width="902" alt="screenshot1" src="https://github.com/clangd/clangd/assets/25045666/ffc4c38b-6bb9-470d-ac59-11b5ce01b3aa">
<img width="902" alt="screenshot2" src="https://github.com/clangd/clangd/assets/25045666/09d8fd4f-a77d-40fa-af98-9d0eefffc937">

**Reply by HighCommander4** [reply]
The first screenshot is quite surprising: it shows that a sequence of 18 characters containing 15 spaces followed by `ene` (which is the beginning of an identifier token) is getting a semantic token type of `macro`.

This looks like a clangd bug to me. Unfortunately, I'm not sure how to investigate it further without having a reproducing code example. If you're able to share a reproducing code example (even if it's not minimal), that would be great.

**Reply by ws2bhhh** [reply]
Hoping my poor level of coding doesn't bother you.😔
[code.txt](https://github.com/clangd/clangd/files/12519834/code.txt)
